A quick fix to get past seg faults.

jwscook opened the pull request:
The c code was expecting 32 bit integers instead of 1 bit integers (`Bool`s) for boolean information. Hence pointers of julia structs containing `Bool`s were the wrong length. This is a quick fix in that the julia structs now contain `Cint`s and not `Bool`s, and associated `if` statements have been changed to reflect the fact that fields are now of type `Cint` e.g. `if struct.foo::Bool ...` -> `if struct.foo::Cint == true ...`.

Commit:
fix run unzip at least on my machine

Commit:
use Actions

Commit:
rolled back version of CSV, set GC safepoints in tests to prevent segfaults

Commit:
Project and Manifest tomls

Commit:
Project and Manifest tomls

Commit:
remove GC calls in runtests

Commit:
remove GC calls in runtests

Commit:
converted bools to ints and altered if statements

Commit:
Merge branch 'master' of github.com:jwscook/JuliaFMI

jwscook commented on Project.toml:
```diff
@@ -6,6 +6,7 @@ version = "0.0.1"
 [deps]
 CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+Debugger = "31a5f54b-26ea-5ae9-a837-f05ce5417438"
```
I forgot to remove this after a debugging sessions

jwscook commented on src/FMUSimulator.jl:
```diff
@@ -284,7 +284,7 @@ function my_unzip(target::String, destinationDir::String)
     try
         #use unzip
-        run(Cmd(`unzip -qo $target`, dir = destinationDir))
+        run(`unzip -qo $target -d $destinationDir`)
```
I changed this so it would run on ubuntu - not sure if it runs on Windows; the CI will tell.

Commit:
re-added CSV to manifest to swap git@ to https:// in repo-url

Commit:
add CSV v0.6 to compat in Project.toml

jwscook committed:
ci-workflow.yml julia 1.6 down to 1.5

AnHeuermann commented:
I like your changes to src/FMI2Types.jl.
`const CintThatIsActuallyABoolean = Cint` fits very well in my verbose naming scheme 😅 

Could you confirm the bug from #1 when uncommenting https://github.com/AnHeuermann/JuliaFMI/blob/master/src/FMUSimulator.jl#L634 and running the test before your changes? On julia-1.5.3 (Linux) it doesn't seem to crash any more, at least for me. But I think it was mostly a Windows problem any way.

Also I would like to separate the CI changes into a different PR. Before I was using Travis, but if you get the GitHub workflow running on your master branch I have no problem with changing to workflow.
Btw. I have the same problem as in https://github.com/jwscook/JuliaFMI/runs/1832087914
I had to install LightXML manually to precompile [JuliaFMI](https://github.com/JuliaIO/LightXML.jl).

Commit:
remove actions yml from PR

Commit:
actions fixup

jwscook commented:
Actions CI is failing with the same error message as https://github.com/julia-actions/julia-runtest/issues/18, so I'll need to check for circular dependencies. 

> Could you confirm the bug from #1 when uncommenting https://github.com/AnHeuermann/JuliaFMI/blob/master/src/FMUSimulator.jl#L634 and running the test before your changes? On julia-1.5.3 (Linux) it doesn't seem to crash any more, at least for me. But I think it was mostly a Windows problem any way.

I've uncommented that line and run it with 1.6 (on Ubuntu 20.04.2 LTS) and julia didn't seg fault during tests as before - not all the tests pass though, but it's a step in the right direction. I tried running it on a fresh 1.5 and got this error:

```bash
┌─05-Feb 13:31:41 J0731 ~/Documents/code/julia/JuliaFMI/test (branch:  ) 
└─$julia1.5 --project=..
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.4-pre.0 (2020-11-10)
 _/ |\__'_|_|_|\__'_|  |  release-1.5/599ecd8210 (fork: 179 commits, 273 days)
|__/                   |

(JuliaFMI) pkg> instantiate
ERROR: AssertionError: sourcepath !== nothing
Stacktrace:
 [1] is_package_downloaded at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:1670 [inlined]
 [2] #12 at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:113 [inlined]
 [3] _all(::Pkg.Operations.var"#12#13"{Pkg.Types.Context}, ::Array{Pkg.Types.PackageSpec,1}, ::Colon) at ./reduce.jl:828
 [4] #all#638 at ./reducedim.jl:735 [inlined]
 [5] all at ./reducedim.jl:735 [inlined]
 [6] is_instantiated at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:113 [inlined]
 [7] instantiate(::Pkg.Types.Context; manifest::Nothing, update_registry::Bool, verbose::Bool, platform::Pkg.BinaryPlatforms.Linux, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:828
 [8] instantiate(::Pkg.Types.Context) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:795
 [9] #instantiate#169 at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:791 [inlined]
 [10] instantiate() at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:791
 [11] do_cmd!(::Pkg.REPLMode.Command, ::REPL.LineEditREPL) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/REPLMode/REPLMode.jl:401
 [12] do_cmd(::REPL.LineEditREPL, ::String; do_rethrow::Bool) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/REPLMode/REPLMode.jl:382
 [13] do_cmd at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/REPLMode/REPLMode.jl:377 [inlined]
 [14] (::Pkg.REPLMode.var"#24#27"{REPL.LineEditREPL,REPL.LineEdit.Prompt})(::REPL.LineEdit.MIState, ::Base.GenericIOBuffer{Array{UInt8,1}}, ::Bool) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/Pkg/src/REPLMode/REPLMode.jl:546
 [15] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [16] invokelatest at ./essentials.jl:709 [inlined]
 [17] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2355
 [18] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /home/cookj/builds/julia-1.5/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1144
 [19] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:356

julia> 
                                                                                             [ 22s562 | Feb 05 13:34:26 > Feb 05 13:34:49 ]
┌─05-Feb 13:34:49 J0731 ~/Documents/code/julia/JuliaFMI/test (branch:  ) 
└─$julia1.5 --project=.. runtests.jl 
ERROR: LoadError: ArgumentError: Package Zlib_jll does not have JLLWrappers in its dependencies:
- If you have Zlib_jll checked out for development and have
  added JLLWrappers as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with Zlib_jll
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:906
 [2] include(::Function, ::Module, ::String) at ./Base.jl:380
 [3] include(::Module, ::String) at ./Base.jl:368
 [4] top-level scope at none:2
 [5] eval at ./boot.jl:331 [inlined]
 [6] eval(::Expr) at ./client.jl:467
 [7] top-level scope at ./none:3
in expression starting at /home/cookj/.julia/packages/Zlib_jll/BGVLi/src/Zlib_jll.jl:5
ERROR: LoadError: LoadError: Failed to precompile Zlib_jll [83775a58-1f1d-513f-b197-d71354ab007a] to /home/cookj/.julia/compiled/v1.5/Zlib_jll/xjq3Q_94zM6.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include(::Module, ::String) at ./Base.jl:368
 [8] top-level scope at /home/cookj/.julia/packages/JLLWrappers/WnWcZ/src/toplevel_generators.jl:170
 [9] include(::Function, ::Module, ::String) at ./Base.jl:380
 [10] include(::Module, ::String) at ./Base.jl:368
 [11] top-level scope at none:2
 [12] eval at ./boot.jl:331 [inlined]
 [13] eval(::Expr) at ./client.jl:467
 [14] top-level scope at ./none:3
in expression starting at /home/cookj/.julia/packages/XML2_jll/Slt3Q/src/wrappers/x86_64-linux-gnu.jl:4
in expression starting at /home/cookj/.julia/packages/XML2_jll/Slt3Q/src/XML2_jll.jl:8
ERROR: LoadError: Failed to precompile XML2_jll [02c8fc9c-b97f-50b9-bbe4-9be30ff0a78a] to /home/cookj/.julia/compiled/v1.5/XML2_jll/xWVg2_94zM6.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include(::Module, ::String) at ./Base.jl:368
 [8] top-level scope at none:2
 [9] eval at ./boot.jl:331 [inlined]
 [10] eval(::Expr) at ./client.jl:467
 [11] top-level scope at ./none:3
in expression starting at /home/cookj/.julia/packages/LightXML/g6rbQ/src/LightXML.jl:3
ERROR: LoadError: LoadError: Failed to precompile LightXML [9c8b4983-aa76-5018-a973-4c85ecc9e179] to /home/cookj/.julia/compiled/v1.5/LightXML/a89nj_94zM6.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include at ./Base.jl:368 [inlined]
 [8] include(::String) at /home/cookj/Documents/code/julia/JuliaFMI/src/JuliaFMI.jl:6
 [9] top-level scope at /home/cookj/Documents/code/julia/JuliaFMI/src/JuliaFMI.jl:10
 [10] include(::Function, ::Module, ::String) at ./Base.jl:380
 [11] include(::Module, ::String) at ./Base.jl:368
 [12] top-level scope at none:2
 [13] eval at ./boot.jl:331 [inlined]
 [14] eval(::Expr) at ./client.jl:467
 [15] top-level scope at ./none:3
in expression starting at /home/cookj/Documents/code/julia/JuliaFMI/src/modelDescriptionParsing.jl:6
in expression starting at /home/cookj/Documents/code/julia/JuliaFMI/src/JuliaFMI.jl:10
ERROR: LoadError: LoadError: Failed to precompile JuliaFMI [32430f9e-34ed-11e9-2e7d-ffb2dc54003b] to /home/cookj/.julia/compiled/v1.5/JuliaFMI/HvBJh_94zM6.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
 [6] include(::String) at ./client.jl:457
 [7] top-level scope at /home/cookj/Documents/code/julia/JuliaFMI/test/runtests.jl:12
 [8] include(::Function, ::Module, ::String) at ./Base.jl:380
 [9] include(::Module, ::String) at ./Base.jl:368
 [10] exec_options(::Base.JLOptions) at ./client.jl:296
 [11] _start() at ./client.jl:506
in expression starting at /home/cookj/Documents/code/julia/JuliaFMI/test/tests.jl:6
in expression starting at /home/cookj/Documents/code/julia/JuliaFMI/test/runtests.jl:12
                                                                                              [ 8s434 | Feb 05 13:34:52 > Feb 05 13:35:01 ]
```

Incidentally that's the same error that Actions CI is getting so I'll take a look at that. Anyway, I added `LightXML` before trying to instantiate:

```bash
┌─05-Feb 13:35:01 J0731 ~/Documents/code/julia/JuliaFMI/test (branch:  ) 
└─$julia1.5 --project=..
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.4-pre.0 (2020-11-10)
 _/ |\__'_|_|_|\__'_|  |  release-1.5/599ecd8210 (fork: 179 commits, 273 days)
|__/                   |

(JuliaFMI) pkg> add LightXML
  Resolving package versions...
  Installed DataAPI ─ v1.5.1
No Changes to `~/Documents/code/julia/JuliaFMI/Project.toml`
Updating `~/Documents/code/julia/JuliaFMI/Manifest.toml`
  [0dad84c5] - ArgTools
  [56f22d72] + Artifacts v1.3.0
  [e66e0078] + CompilerSupportLibraries_jll v0.3.4+0
  [9a962f9c] ↑ DataAPI v1.5.0 ⇒ v1.5.1
  [f43a241f] - Downloads
  [b27032c2] - LibCURL
  [deac9b47] - LibCURL_jll
  [29816b5a] - LibSSH2_jll
  [c8ffd9c3] + MbedTLS_jll v2.16.8+1
  [14a3606d] - MozillaCACerts_jll
  [ca575930] - NetworkOptions
  [fa267f1f] - TOML
  [a4e569a6] - Tar
  [02c8fc9c] ↓ XML2_jll v2.9.11+0 ⇒ v2.9.10+3
  [83775a58] + Zlib_jll v1.2.11+18
  [8e850ede] - nghttp2_jll

(JuliaFMI) pkg> instantiate

(JuliaFMI) pkg> 
```

and then it ran in the same way as on 1.6.

> Also I would like to separate the CI changes into a different PR. 

Yup.

Commit:
removed Debugger from project

Commit:
added test target to Project.toml

jwscook reviewed (commented):
I've dealt with these two items.

jwscook commented:
@AnHeuermann are you happy with this to be merged in?

AnHeuermann reviewed (approved):
Looks god to me
